packages/animations/lib/src/utils/composite_animation_widget.dart

```diff
@@ -0,0 +1,253 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:animations/animations.dart';
+import 'package:flutter/material.dart';
+
+/// Signature for a function that creates a widget that builds a
+/// enter transition.
+///
+/// Used by [CompositeAnimationWidget]
+typedef _EnterTransitionBuilder = Widget Function(
+    BuildContext, Animation<double>, Widget);
+
+/// Signature for a function that creates a widget that builds a
+/// exit transition.
+///
+/// Used by [CompositeAnimationWidget]
+typedef _ExitTransitionBuilder = Widget Function(
+    BuildContext, Animation<double>, Widget);
+
+/// A widget that animate it's child with different transitions when the
+/// [CompositeAnimationWidget.animation] value goes forward
```

Comment:
You shouldn't need the `CompositeAnimationWidget.` here, dartdoc should figure this out on its own.

```suggestion
/// [animation] value goes forward
```

(here and everywhere else in this block of text)